apps/prairielearn/src/ee/pages/instructorInstanceAdminBilling/instructorInstanceAdminBilling.html.ts

```diff
@@ -51,19 +50,23 @@ export function InstructorCourseInstanceBilling({
           <div class="card mb-4">
             <div class="card-header bg-primary text-white d-flex">Billing</div>
             <div class="card-body">
-              ${InstructorInstanceAdminBillingForm({
-                initialRequiredPlans: requiredPlans,
-                desiredRequiredPlans: requiredPlans,
-                institutionPlanGrants,
-                courseInstancePlanGrants,
-                enrollmentCount,
-                enrollmentLimit,
-                enrollmentLimitSource,
-                externalGradingQuestionCount,
-                workspaceQuestionCount,
-                editable,
-                csrfToken: resLocals.__csrf_token,
-              })}
+              ${renderWithProps(
+                'InstructorInstanceAdminBillingForm',
+                InstructorInstanceAdminBillingForm,
```

Comment:
This is the function call that Astro more or less does "magically".

Reply:
This doesn't feel very heavy-weight to me, but I guess we were already doing something basically identical so I'm already conditioned to do this. How would this look with Astro, as a matter of curiosity?

Reply:
See https://docs.astro.build/en/guides/framework-components/#hydrating-interactive-components. You essentially just use a component as normal with a `client:` directive, and Astro then magically figures out which JS to bundle and what data to serialize, and automatically does the hydration on the client.